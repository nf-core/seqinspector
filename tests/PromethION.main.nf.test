nextflow_pipeline {

    name "Test Workflow main.nf on PromethION data"
    script "../main.nf"
    tag "seqinspector"
    tag "PIPELINE"

    test("PromethION data test") {

        when {
            config "./PromethION.main.nf.test.config"
            params {
                outdir = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    path("$outputDir/multiqc/global_report/multiqc_data/multiqc_citations.txt"),
                    path("$outputDir/multiqc/global_report/multiqc_data/multiqc_fastqc.txt"),
                    path("$outputDir/multiqc/global_report/multiqc_data/multiqc_general_stats.txt"),
                    path("$outputDir/multiqc/global_report/multiqc_data/multiqc_software_versions.txt"),
                    ).match()
                },
            )
        }
    }
}
