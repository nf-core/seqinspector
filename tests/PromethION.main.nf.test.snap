{
    "PromethION data test": {
        "content": [
            "multiqc_citations.txt:md5,4c806e63a283ec1b7e78cdae3a923d4f",
            "multiqc_fastqc.txt:md5,35984961f25a0d4e7352cab4d5650178",
            "multiqc_general_stats.txt:md5,1465b0b1959e3864b28ecc2340df351b",
            "multiqc_software_versions.txt:md5,49e3596d49ee49d967d3b6c363b486d5"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-10T08:58:57.180636"
    }
}